### DISCUSSION TITLE
Unknown config parameter with bluetooth.config()

### DISCUSSION THREAD

**Discussion by Teegrammer** [micropython/micropython#12800] [Libraries & Drivers] [answered]
Hi,
i'm trying to use BLE for pairing my Pico W with another device (with `BLE.gap_pair(conn_handle, /)`) and sending some data. As noted in the documentation i tried setting the io, mitm, le_secure and bond settings with `bluetooth.config()`. Unfortunatly i always get an unknown config parameter error. I also tried setting other parameters, there i am able to change mtu for example. To make sure my code isn't the issue i tried the example script **ble_bonding_peripheral.py** with the same errors.

On another note: I tried using `BLE.gap_pair(conn_handle, /)` without the parameters, but even that function seems to undefined (haven't found it in **bluetooth.pyi**). So far i haven't had issues with any other part of micropython. I'm using version 1.21.0 and used `sys.implementation` to confirm it. 

Thank you in advance for your help :)
Tee

**Comment by jimmo** [comment, marked as answer]
@Teegrammer pairing & bonding is not yet supported on the Pico W. Hopefully this will be in 1.22.

**Reply by Teegrammer** [reply]
Oh ok, thank you for your answer. Could you guide me to where i can find this information in the documentation so that i don't run into a similar issue in the future?

**Reply by jimmo** [reply]
It's here https://docs.micropython.org/en/latest/library/bluetooth.html#pairing-and-bonding

> Note: This is currently only supported when using the NimBLE stack on STM32 and Unix (not ESP32).

Should probably add Pico/RP2040 alongside ESP32 there.

**Comment by brianreinhold** [comment]
@jimmo I am trying to fix the distirbution so pairing/bonding is supported on the PICO-W. Is it just a matter of setting some of the defines (because it is supported in the C-SDK). I've tried one setting (in the CMAKE file for the PICO-W but it did not work, Now I am simply changing the defaults and checking that. I dont get the config errors anymore but I do know that is not the correct place to set the parameters as it applies to all platforms. How (and where) should I do it?

**Reply by brianreinhold** [reply]
Not it is not just a matter of setting defines. There are missing handlers for the btstack's security manager. This will take some effort!
